.github/workflows/build-cli.yml

```diff
@@ -0,0 +1,23 @@
+name: "Build CLI"
+on: [push]
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Setup .NET Core SDK 6
+        uses: actions/setup-dotnet@v1
+        with:
+          dotnet-version: 6.x
+      - uses: actions/cache@v2
+        with:
+          path: ~/.nuget/packages
+          # Look to see if there is a cache hit for the corresponding requirements file
+          key: ${{ runner.os }}-nuget-${{ hashFiles('**/packages.lock.json') }}
+          restore-keys: |
+            ${{ runner.os }}-nuget
+      - name: Install dependencies
```

Comment:
because the kiota packages are not published yet, you're going to need a nuget config + a token, have a look at the kiota workflow for that.
https://github.com/microsoft/kiota/blob/8449c9e67101de683ab44c99ec2fb00bd7bd59c0/.github/workflows/http-dotnet-httpclient.yml#L23